packages/controllers/src/plugins/PluginController.ts

```diff
@@ -928,16 +935,36 @@ export class PluginController extends BaseController<
    *
    * @param pluginName - The name of the plugin whose message handler to get.
    */
-  async getRpcMessageHandler(pluginName: string) {
-    const handler = await this._getRpcMessageHandler(pluginName);
-    if (!handler) {
-      return undefined;
+  async getRpcMessageHandler(
+    pluginName: string,
+  ): Promise<
+    (origin: string, request: Record<string, unknown>) => Promise<unknown>
+  > {
+    const existingHandler = this._rpcHandlerMap.get(pluginName);
+    if (existingHandler) {
+      return existingHandler;
     }
 
-    return (origin: string, request: Record<string, unknown>) => {
+    const rpcHandler = async (
+      origin: string,
+      request: Record<string, unknown>,
+    ) => {
+      let handler = await this._getRpcMessageHandler(pluginName);
+      if (!handler) {
+        // cold start
+        if (this.isRunning(pluginName) === false) {
+          await this.startPlugin(pluginName);
+          handler = await this.getRpcMessageHandler(pluginName);
```

Comment:
I suspect we're not trying to recurse here 😅 

```suggestion
          handler = await this._getRpcMessageHandler(pluginName);
```